### DISCUSSION TITLE
Can I run containerlab in my Raspberry pi 4 8gb?

### DISCUSSION THREAD

**Discussion by rodhpadilla** [srl-labs/containerlab#897] [Q&A] [answered]
Does anyone know if containerlab support ARM64 processors?  
in the documentation only mention about AMD64

**Comment by hellt** [comment]
Hi
we do not build arm64 containerlab, since there is a shortage of network OS that support this arch
but you can build one yourself for sure

**Comment by rodhpadilla** [comment]
Hey Roman, thanks a lot for your quick answer!
To be honest with you I have not idea about how can I do this. 
Any idea/recommendation about how can I start reading about build this for ARM or any other architecture.

**Comment by hellt** [comment]
you just need to instal Go on your system and then run `go build` command
this article should give all the pointers
https://www.digitalocean.com/community/tutorials/how-to-build-and-install-go-programs

which network os you want to use containerlab with on rpi?

**Reply by rodhpadilla** [reply]
it would be great CSR1000v/XRv9k or whatever tiny network OS that does not require too much ram. In the past I did use vyattaOS and it went OK in an old linux box.

**Reply by hellt** [reply]
those images are only available for amd64 arch, so you won't be able to run them on rpi
I don't know if vyatta has an arm build

**Reply by rodhpadilla** [reply]
I think VyOS has an ARM build 
https://docs.vyos.io/en/crux/contributing/build-vyos.html

**Comment by hellt** [comment, marked as answer]
Linux/Arm64 builds for containerlab have been provided

**Reply by rodhpadilla** [reply]
this is great Roman!! thank you
I'll check it in a few weeks
